core/backend/src/rpc-impl/IModelReadRpcImpl.ts

```diff
@@ -119,6 +119,17 @@ export class IModelReadRpcImpl extends RpcInterface implements IModelReadRpcInte
     return viewHydrater.getHydrateResponseProps(options);
   }
 
+  public async hydrateDefaultViewState(tokenProps: IModelRpcProps): Promise<HydrateViewStateResponseProps> {
```

Comment:
While we were discussing in person, you mentioned that there was a chance 'getCustomViewState3dData' returned the modelIds in a different order which then lead to the hydrateViewState request from the frontend requesting information on the models in a different order. This reduces the cacheability which is the motivation for this PR I believe (Please correct me if I'm wrong). Could you give more details on that?

The backend calls sortAndCompress on the modelIds before returning them to the frontend, but you're saying calling sortAndCompress on the same modelIds can sometimes yield different results?

Reply:
It's not sortAndCompress that's the problem - it's that the marshaling/unmarshaling of the rpc system is (most likely) the problem. For example, I have two apps that will call a backend - one through the rpc system, the other through a standard HTTP request. The impl for marshaling (or serialization) of the rpc request parameters are transparent and viewable within this repo, so I follow that same impl for my HTTP request.  And yet, they produce two **near-identical** serialized parameters, but they are not identical - thus, preventing the caching capabilities. 
![image](https://github.com/user-attachments/assets/05949bae-5bab-4c20-ab3e-5dd279f26b17)
In the image above, I highlighted the discrepeancy between the two.